zip-0315.rst

```diff
@@ -4,6 +4,315 @@
   Title: Best Practices for Wallet Handling of Multiple Pools
   Owners: Daira Hopwood <daira@electriccoin.co>
           Jack Grigg <jack@electriccoin.co>
-  Status: Reserved
+  Status: Proposed
   Category: Wallet
   Discussions-To: <https://github.com/zcash/zips/issues/447>
+  Pull-Request: <https://github.com/zcash/zips/issues/>
+
+
+Terminology
+===========
+
+The key words "MUST", "SHOULD", "SHOULD NOT", and "MAY" in this document are to be
+interpreted as described in RFC 2119. [#RFC2119]_
+
+The terms below are to be interpreted as follows:
+
+Auto-shielding
+    The process of automatically transferring transparent funds to the most recent
+    shielded pool on receipt.
+
+Auto-migration
+    The process of automatically transferring shielded funds from older pools to the
+    most preferred (usually the most recent) pool on receipt.
+
+TODO: add definitions of opportunistic shielding and migration.
+
+TODO: "confirmed", "partially confirmed"
+
+
+Motivation
+==========
+
+The aim of this ZIP is to provide wallet developers with a set of best practices by
+which they can minimize the leakage of information when constructing transactions.
+
+This includes best practices for:
+
+* how to handle interactions between the ZIP 32 key tree and Unified Addresses;
+* when to use external or internal keys/addresses;
+* sharing addresses and viewing keys;
+* sending and receiving funds;
+* migrating funds between pools.
+
+
+Requirements
+============
+
+User consent
+------------
+
+The guiding principle of these requirements is that users must explicitly consent
+to each instance of information being revealed by the wallet.
+
+As is true for consent in general, a user may give blanket consent to reveal a
+particular kind of information, and must always be able to revoke consent to
+reveal such information in the future. The specifications below describe some
+situations in which blanket consent may be inappropriate.
+
+
+Long-term storage of funds
+--------------------------
+
+It is RECOMMENDED that wallets only hold funds as shielded in the long-term;
+that is, they automatically shield incoming transparent funds.
```

Comment:
```suggestion
that is, if a wallet supports receiving transparent funds, then it SHOULD
automatically shield such funds.
```

Reply:
Add some motivation either here or in the Motivation section.

E.g. the note anonymity set over time is maximized.